### PR TITLE
chore: Phone Input Widget Integration

### LINKED ISSUE #27746: Add functionality to WDS TextInput to support PhoneInput widget
When adding functionality, follow the [audit](https://www.notion.so/appsmith/Widgets-Properties-Audit-3832f6a7be5f47cc9bfa3bd1ddae31fa)

### PR DESCRIPTION
**Pull request by jsartisan**
Fixes #27746

### CODE REVIEW

**Comment by ichik** [comment]
@jsartisan ”Change country code” doesn't seem to work.

File: app/client/src/widgets/wds/WDSPhoneInputWidget/widget/config/propertyPaneConfig/validations/defaultValueValidation.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import type { PhoneInputWidgetProps } from "../../../types";
+
+export function defaultValueValidation(
+  value: any,
+  props: PhoneInputWidgetProps,
+  _?: any,
```
Comment (KelvinOm) [inline app/client/src/widgets/wds/WDSPhoneInputWidget/widget/config/propertyPaneConfig/validations/defaultValueValidation.ts:6]: Can we get rid of `any`?

File: app/client/src/widgets/wds/WDSPhoneInputWidget/widget/config/propertyPaneConfig/validations/defaultValueValidation.test.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import type { PhoneInputWidgetProps } from "../../../types";
+import { defaultValueValidation } from "./defaultValueValidation";
+import _ from "lodash";
+
+describe("defaultValueValidation", () => {
+  let result: any;
```
Comment (KelvinOm) [inline app/client/src/widgets/wds/WDSPhoneInputWidget/widget/config/propertyPaneConfig/validations/defaultValueValidation.test.ts:6]: Can we set correct type here?

**Review by KelvinOm — APPROVED** [review]

**Comment by jsartisan** [comment]
> ”Change country code” doesn't seem to work.
@ichik Yes, that will be handled after we create the select component.

**Review by KelvinOm — APPROVED** [review]

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**